### ISSUE TITLE
No command handlers were found for "MyCommand"

### ISSUE THREAD

**Issue by matheusdsl** [tyronegroves/SimpleCQRS#31]
Hello,

I'm having an issue where my project suddenly stops handling commands.
The exception thrown says "No command handlers were found for "MyCommand"".

It is a .Net Web API 2.0 project. It builds fine, runs fine for some time (we managed to run it for something like 2 days straight),
but it suddenly stops handling the commands.
The only way to make it work is by publishing it again or cleaning the solution (when we are running it on debugging mode).

Any help would be appreciated.

```

public class MyController : ApiController
{
        public ICommandBus CommandBus;
        public MyController() : this(ServiceLocator.Current.Resolve<ICommandBus>())
        {
        }
        public MyController(ICommandBus commandBus)
        {
            CommandBus = commandBus;
        }
        public void Execute()
        {
            MyCommand cmd = new MyCommand();
            CommandBus.Execute(cmd);
        }
}

public class MyCommand : CommandWithAggregateRootId
{
        public string Test { get; set; }
}

public class MyCommandHandler : AggregateRootCommandHandler<MyCommand, MyDomain>
{
        protected IDomainRepository _repository;

        public MyCommandHandler(IDomainRepository repository)
        {
            _repository = repository;
        }

        public override void Handle(MyCommand command, MyDomain aggregateRoot)
        {
            if (aggregateRoot == null)
                aggregateRoot = new MyDomain();

            aggregateRoot.Execute(command.Test);

            _repository.Save(aggregateRoot);
        }
 }

```

**Comment by tyronegroves** [comment]
Are here dynamic assemblies being generated? The service locator may be looking at a type in a specific assembly then that assembly changed or domain unloaded. Just a thought.

**Comment by matheusdsl** [comment]
There are no dynamic assemblies, just static.

We could solve the issue, however, by adding Nuget packages for Ninject. We had another project in our solution which was using it and we noticed this issue didn't occur with it.
We then compared the packages and noticed that Ninject was missing in the project with the issue.
Not sure how it could be related but apparently it solved the problem. 
Since we added Ninject we've been running the project for 4 days non-stop.

Anyways, thank you for the help!
